src/actions/PermissionsActions.js

```diff
@@ -9,26 +9,32 @@ import {
 
 import {permissionResponses} from '../constants';
 
-const {DENIED} = permissionResponses;
+const {GRANTED, DENIED} = permissionResponses;
+
+const getLocation = async (dispatch) => {
+
+    try {
+        const location = await Location.getCurrentPositionAsync({enableHighAccuracy: true});
+        const {coords: {longitude, latitude}} = location;
+
+        return dispatch({
+            type: CURRENT_LOCATION_UPDATED,
+            payload: {currentLocation: {longitude, latitude}}
+        })
+
+    } catch (error) {
+        console.log(error);
+        return dispatch({
+            type: LOCATION_SERVICES_PERMISSION_UPDATED,
+            payload: {locationServicesPermission: DENIED}
+        })
+    }
```

Comment:
Put `getLocation` into its own function, so that we can both call it when we explicitly call `getCurrentUserLocation`, but also when we `promptForLocationServicesPermission` and realize that they've already `GRANTED` us access. This was the fix for GH-52